crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1283,7 +1366,14 @@ impl<'db> Type<'db> {
             Type::ModuleLiteral(_) => KnownClass::ModuleType.to_class(db),
             Type::Tuple(_) => KnownClass::Tuple.to_class(db),
             Type::ClassLiteral(ClassLiteralType { class }) => class.metaclass(db),
-            // TODO can we do better here? `type[LiteralString]`?
+            Type::SubclassOf(SubclassOfType { class }) => Type::SubclassOf(
+                class
+                    .try_metaclass(db)
+                    .ok()
+                    .and_then(Type::into_class_literal)
+                    .unwrap_or(KnownClass::Type.to_class(db).expect_class_literal())
+                    .to_subclass_of_type(),
```

Comment:
This looks great, too! Falling back to `Type::SubclassOf(builtins.type)` in the case of any non-specific metaclass is the right answer here, I think.